### ISSUE TITLE
Is it possible to reveal the code for tissue compositions/ground truth compositions estimation?

### ISSUE THREAD

**Issue by pwwang** [jasminezhoulab/cfSort#5]

**Comment by DashuoLI** [comment]
Hi, 

The code for estimating tissue compositions is in the codes folder. This folder includes all codes from the input methylation features to the tissue composition estimate.

**Comment by pwwang** [comment]
Hi, thanks for the prompt reply. I suppose you mean `data_prep.py`. However, it uses `label_filename` (e.g demo/input.label.txt) as input, which contains the ground truth compositions already. As described in the paper, they are generated by 4 steps, including the simulation of methylation data. How to generate the compositions in the label file is what I am looking for.

**Comment by DashuoLI** [comment]
Hi, 

Thanks for your interest. If you are working on deconvolution of mixture samples, please use `data_prep.for_test_only.py`. This is the script to prepare testing samples for estimating tissue compositions. It does not require `label_filename`. 

The tissue compositions for training data were generated randomly. Once a composition was generated, we would generate a corresponding training sample by mixing the DNA fragments from different tissue types based on this composition. Then the randomly generated tissue composition and the corresponding training sample were used to build the DNN model. Therefore, there is no tissue composition estimated for the training data.
